### ISSUE TITLE
is it possible to create a global var that's shared between pages?

### ISSUE THREAD

**Issue by bhurlow** [zeit/next.js#203]

**Comment by impronunciable** [comment]
```
// myglobal.js

export default 42
```

```
// pages/a.js

import glob from '../myglobal'

glob // 42
```


```
// pages/b.js

import glob from '../myglobal'

glob // 42
```

**Comment by ugiacoman** [comment]
Isn't this what props are for?

**Comment by malixsys** [comment]
See the redux example

**Comment by nkzawa** [comment]
> // myglobal.js
>
> export default 42

btw, this wouldn't work on client, when the value was mutated.
Even if two pages requires a same module, the module becomes two different objects on each page, because of code splitting.

**Comment by CompuIves** [comment]
The Redux example is an example with one page. I think the initial question is if it's able to switch between different pages while keeping a global persistent state. I don't think that's possible yet, but may be possible after https://github.com/zeit/next.js/issues/25(?)

**Comment by sedubois** [comment]
@CompuIves, @malixsys is right, the Redux example shows how to keep state across page changes, through the `window` object (although it's quite hacky, IMHO). Similarly for Apollo: https://github.com/sedubois/realate/blob/master/containers/Apollo.js (see https://github.com/zeit/next.js/issues/106).

IMHO a better central management of state is needed, as these issues keep popping up.

**Comment by bhurlow** [comment]
something like a registry API might be appropriate for this e.g.

```
import Registry from 'next'
let session = await Registry.get('session')
```

**Comment by bhurlow** [comment]
Essentially looking for a way to get around the code splitting to store application wide settings. Will try extending the window object but would much prefer a server backed solution instead

**Comment by malixsys** [comment]
You can do all the server-side you want, although a bit difficult at this moment until #25 
just check for `req === true`

**Comment by eezing** [comment]
@bhurlow

Just to clarify:

1. Can you give a real world example of an application wide setting?

2. Do application wide settings ever change durning runtime? If so, do they persist across runtime sessions?

**Comment by bhurlow** [comment]
@eezing sure, 

1. I suppose the obvious example is a session object stored in memory. In a traditional app a global could be referenced and potentially changed from multiple places, but the code splitting (⚡️ )  of next makes this more obscured. Or even something more trivial like a global counter value shared between pages... 

2. no persistence, just talking about the lifespan of the node.js process 

Like @malixsys mentioned though, I can leverage the getInitialProps here and be just fine until #25 

however it could be interesting to expose access to globals from the client as well, but that might be another can of worms

**Comment by arunoda** [comment]
Guys, this is a pretty ugly bug and we fixed this with `v1.2.x`.
Here's a an example app: https://github.com/zeit/next.js/tree/master/examples/shared-modules

Hosted app: https://shared-modules-pjlwifargg.now.sh/
